### ISSUE TITLE
MPV no on brew?

### ISSUE THREAD

**Issue by misiekkb** [mpv-player/mpv#6991]
I get information no formula mpv when I want install from brew? Only on cask brew I may install, this is temporary no mpv on brew repository?

**Comment by aerobounce** [comment]
It seems so. You may want to refer: https://github.com/Homebrew/homebrew-core/commit/41444d526c40b93069b7f0c5414539deb0534179 for information.

And, I've made HEAD only formula for myself.
https://github.com/aerobounce/homebrew-mpv-HEAD
Hope this helps you too.

**Comment by wm4** [comment]
That's a homebrew issue. Apparently 0.29 being outdated caused this. Maybe they'll add it back when there's 0.30.

**Comment by vitorgalvao** [comment]
> Apparently 0.29 being outdated caused this. Maybe they'll add it back when there's 0.30.

That’s unlikely, due to [the reasons given](https://github.com/Homebrew/homebrew-core/commit/41444d526c40b93069b7f0c5414539deb0534179):

> there exists a 'brew cask install mpv', and that the mpv developers do not support tagged releases anymore but only operate from master, and that homebrew maintainer policy regarding tagged releases and the mpv developer policy are in conflict

You may want to consider rehosting the old `homebrew-mpv` repo. More details on https://github.com/mpv-player/mpv/pull/7008.

**Comment by wm4** [comment]
mpv didn't stop tagged releases. It's quite a thing that homebrew insists on using tagged releases only, but then doesn't own up to it and maintains its packages properly.

**Comment by vitorgalvao** [comment]
@wm4 Note that while I am a Homebrew (HB) member and an *avid* mpv user, I had no bearing on the decision. I deal almost exclusively with the Cask (HBC) side; we have different rules and respect the judgements from each side.

I know the distinction between HB and HBC can be a tad confusing (different projects that later merged). In reductive terms, if a piece of software produces an app bundle (i.e. is not CLI-only) then it generally fits in HBC. However, there is still some duplication between projects that we need to deal with slowly.

I’m guessing that there being an issue with the formula (HB) and a ready alternative as a cask (HBC) which is typically where such software would belong anyway (having an app bundle) makes for a compelling reason for removal.

As an mpv user, I meant only to make you aware of the situation *as it stands* and help you find the best way forward, as you may be getting a few open issues on this.

One major difference between HB and HBC is that the former can compile software while the latter cannot, hence why it needs to use the binaries provided by `stolendata`, as referenced in your installation page. I understand some people might not be satisfied by that, considering the difference in popularity between the formula and cask, which is why I mentioned the revival of the `homebrew-mpv` repo.

**Comment by CounterPillow** [comment]
Why is it always Homebrew that makes us have stupid discussions on the issue tracker?

Isn't this like the fourth or fifth time now?

**Comment by vitorgalvao** [comment]
> Why is it always Homebrew that makes us have stupid discussions on the issue tracker?
> 
> Isn't this like the fourth or fifth time now?

I have no idea because I haven’t participated in the others, and like I said I deal with casks, not formulae.

But there’s something that may not yet be clear: **you don’t have to do anything**. *`mpv` still exists on Homebrew*, but instead of `brew install mpv` users `brew cask install mpv`. That’s it. For most users, that will make no difference. However, for some it might. Specifically, for developers you may wish to have a formula that can build from source (casks can’t) or provide a way to install from HEAD.

I came here as an mpv user and fan to provide context on an issue opened by another, so you have all the information to deal with it in the manner you find best. I did so because I like mpv. But I also like Homebrew so I want to help both, not pit them against each other.

You can either take me up on my offer and make specific questions where I’ll do my best to clarify, or you can keep beating the horse. But I opened PRs and made a comment in the name of a productive solution.

**Comment by misiekkb** [comment]
This is big difference between version on mpv and cask. On cask are old version no have example hdr-compute-peak. For me no problem from time to time compilation mpv from file, but this will be very bothersome when I and the people compilation mpv every issue new ffmpeg or mpv, if are add very important thing. Sometimes maybe one of week, sometimes maybe one of day.

I and the people petition to comeback MPV on Homebrew no Cask, please.

**Comment by Akemi** [comment]
macOS doesn't have `hdr-compute-peak` support anyway, because of the lack of compute shaders.

**Comment by aerobounce** [comment]
Isn't the reason mpv formula has been removed exactly like @wm4 said?
> That's a homebrew issue. Apparently 0.29 being outdated caused this. Maybe they'll add it back when there's 0.30.

And on removal commit they say:
> Considering that the tagged release 0.29.1 of mpv cannot be built on macos 10.14.x...

The current situation is, with the latest Xcode, the last tagged mpv cannot be built anymore. That's just the reason I suppose. Which, makes sense.

So why is Swift-related issues resolved commit tagged? mpv is not the project targeted only macOS. That also just make sense.


As a heavy user of mpv, having no CLI mpv is such a big deal. I use it on scripts heavily. With brew (not cask), you can install CLI with app bundle and link it (symlink on PATH) at the the same time automatically.
But such user can just make a repo for themselves just like, until next mpv release comes.

So, I don't think theres issues mpv contributors should care about...?

**Comment by Akemi** [comment]
while not optimal 0.29.1 can still be build on macOS 10.14.x. one has just to disable swift by default and enable it when building with `--HEAD`.

i personally do care to a certain extend. the more people use mpv on macOS the more likely i can fix possible issues with it or become aware of them.

**Comment by aerobounce** [comment]
@Akemi 
> i personally do care to a certain extend. the more people use mpv on macOS the more likely i can fix possible issues with it or become aware of them.
Ah, indeed... as a coder I empathize and I should thank you for all your commitments for macOS side. mpv is literally the best multimedia player available on macOS.

Then, only the solution – the key _right now_ , is tagging..., I assume?

**Comment by vitorgalvao** [comment]
> As a heavy user of mpv, having no CLI mpv is such a big deal.

The cask also provides the CLI.

**Comment by aerobounce** [comment]
> The cask provides the CLI as well.

Oh. And it links, path it naturally. Thanks for the info.
So... the only difference is the binary is where came from. And maybe build flags differences? Lastly, whether contributors are comfortable with that or not.

**Comment by Akemi** [comment]
> Then, only the solution – the key right now , is tagging..., I assume?

if homebrew is not willing to add it back there is nothing we can do. though we will still keep tagging releases (#6936). see my comments on #7008.

**Comment by vitorgalvao** [comment]
> So... for the stable release, the only difference is where the binary is came from. And maybe build flags differences?

Yes. Though I have no idea what the flag differences are, if any. Anecdotally, I’ve noticed some differences between the version installed via the formula and the cask.

The latter (the `stolendata` builds) seems to have worse performance on my machine (videos stutter more often) and custom exit codes seem to not work. I have `q quit-watch-later 9` in my `input.conf`, but when I press <kbd>q</kbd> it exists with an exit code of `0`.

**Comment by vitorgalvao** [comment]
Quick update. I’ve spoken with some of the other maintainers, and we’re open to reinstating `mpv` as a formula if the build issues are resolved. Maybe worth a retry with `0.30`, as mentioned above?

**Comment by wm4** [comment]
These stolendata builds are a year old.

**Comment by vitorgalvao** [comment]
> These stolendata builds are a year old.

So is the latest official tagged released, as you’ve pointed out in another issue. That seems to be root of the whole problem we’re discussing. When there’s a new release, both `stolendata` and Homebrew (assuming it builds as a formula) can update it (even the cask).

**Comment by aerobounce** [comment]
> if homebrew is not willing to add it back there is nothing we can do. though we will still keep tagging releases (#6936). see my comments on #7008.

Yes! #6936. I saw it and I think the idea is absolutely cool.

> Yes. Though I have no idea what the flag differences are, if any. Anecdotally, I’ve noticed some differences between the version installed via the formula and the cask.

I also think brew bottle / HEAD build are somewhat nicer.

---

Current HEAD build has `NSLayoutConstraint` warnings on `macOS Catalina Beta 10 (19A578c)` (does it on latest Mojave too?). Something like this:

```
mpv[20850:304999] This NSLayoutConstraint is being configured with a constant that exceeds internal limits.  A smaller value will be substituted, but this problem should be fixed. Break on BOOL _NSLayoutConstraintNumberExceedsLimit(void) to debug.  This will be logged only once.  This may break in the future.
```

Maybe here? : https://github.com/mpv-player/mpv/blob/802f594a857c703ac88e946d14b69cd3b6eb6006/osdep/macosx_touchbar.m#L194

And yes that's the off topic here to talk about.
I mean, In anyway, macOS Catalina Production release is coming in a few weeks, and it's worth testing on it then think about brew.

**Comment by Akemi** [comment]
> does it on latest Mojave too?

not that i know of. if it's still a problem with the GM/release report it as a separate issue. though i don't know yet when i will be able to update to 10.15

**Comment by Moulick** [comment]
ok now question at the end of 2020, which MPV to install cask or formula ?

**Comment by vitorgalvao** [comment]
> ok now question at the end of 2020, which mpv to install cask or formula ?

The cask provides an app bundle; the formula does not. The formula can install from HEAD, the cask cannot. Make your choice depending on what’s important to you.

Related: https://github.com/Homebrew/homebrew-cask/issues/86226.

**Comment by Moulick** [comment]
Thanks @vitorgalvao , I think I need the cask as I mainly need the app bundle
